ui/webui/src/components/AnacondaWizard.jsx

```diff
@@ -130,160 +130,143 @@ export const AnacondaWizard = ({ dispatch, isBootIso, osRelease, storageData, lo
+        osRelease,
+        setIsFormValid,
+        setStorageEncryption,
+        setStorageScenarioId: (scenarioId) => {
+            window.sessionStorage.setItem("storage-scenario-id", scenarioId);
+            setStorageScenarioId(scenarioId);
+        },
+        stepNotification,
+        storageEncryption,
+        storageScenarioId,
     };
 
-    const createSteps = (stepsOrder) => {
-        const steps = stepsOrder.filter(s => !s.isHidden).map(s => {
-            let step = ({
+    const createSteps = (stepsOrder, componentProps) => {
+        return stepsOrder.filter(s => !s.isHidden).map(s => {
+            let stepProps = {
+                ariaLabel: `${s.label} content`,
                 id: s.id,
                 name: s.label,
                 stepNavItemProps: { id: s.id },
-                canJumpTo: canJumpToStep(s.id, currentStepId),
-                isFinishedStep: isFinishedStep(s.id),
-            });
+                ...(s.isExpandable && { isExpandable: true }),
+            };
             if (s.component) {
-                step = ({
-                    ...step,
-                    component: (
+                stepProps = {
+                    children: (
                         <s.component
                           idPrefix={s.id}
-                          setIsFormValid={setIsFormValid}
-                          onCritFail={onCritFail}
-                          onAddErrorNotification={onAddErrorNotification}
-                          stepNotification={stepNotification}
-                          isInProgress={isInProgress}
-                          storageEncryption={storageEncryption}
-                          setStorageEncryption={setStorageEncryption}
-                          storageScenarioId={storageScenarioId}
-                          isBootIso={isBootIso}
-                          osRelease={osRelease}
-                          setStorageScenarioId={(scenarioId) => {
-                              window.sessionStorage.setItem("storage-scenario-id", scenarioId);
-                              setStorageScenarioId(scenarioId);
-                          }}
+                          {...componentProps}
                           {...s.data}
                         />
                     ),
-                });
+                    ...stepProps
+                };
             } else if (s.steps) {
-                step.steps = createSteps(s.steps);
+                const subSteps = createSteps(s.steps, componentProps);
+                stepProps = {
+                    ...stepProps,
+                    steps: [...subSteps]
+                };
             }
-            return step;
+            return (
+                <WizardStep key={s.id} {...stepProps} />
+            );
         });
-        return steps;
     };
-    const steps = createSteps(stepsOrder);
+    const steps = createSteps(stepsOrder, componentProps);
 
-    const goToStep = (newStep, prevStep) => {
```

Comment:
Do you mind keeping the variable naming how is was before, just to cause less diff in the commit? Let's limit the commit to the required changes. Any improvements can come if you like in follow up, and be discussed separately.